### DISCUSSION TITLE
Possible to integrate CesiumJS into a Dioxus app?

### DISCUSSION THREAD

**Discussion by clintfred** [DioxusLabs/dioxus#386] [Q&A] [answered]
I'm a web dummy, so I hope this question makes sense.

I currently have a [CesiumJS](https://github.com/CesiumGS/cesium) app written using the Typescript definitions. I am primarily a Rust developer and am looking for ways to embed my Cesium UI into a larger application that can work across Web, Desktop, Mobile. Is dioxus a good choice for this type of app? I saw some mention of embedding React components into a dioxus app, but I didn't really see it mentioned in the guide, so maybe that's not really a supported use case? If it is supported, how does communication to/from JS work?

The tool looks really cool. Thanks for any input you are able to provide.

**Comment by jkelleyrtp** [comment, marked as answer]
Hey! I imagine you could integrate Cesium with a Dioxus app.


However looking into their docs, it seems that they rely pretty heavily on NPM to do things like tree shaking. The traditional dioxus project configuration won't work (probably).

I would look at the guide on how to integrate rust into npm projects here:

https://rustwasm.github.io/wasm-pack/book/tutorials/hybrid-applications-with-webpack/index.html
